### DISCUSSION TITLE
Uncaught ReferenceError: $wire is not defined

### DISCUSSION THREAD

**Discussion by jochemgruter** [livewire/livewire#2203] [1. Help]
Everything I have already intergrated with livewire is working fine.  But when I try to access `$wire` variable, I got the error

`Uncaught ReferenceError: $wire is not defined`

I have tried many things. Even with this code I keep saying the error:

```
@livewireScripts

<script>
    document.addEventListener('livewire:load', function () {
        console.log($wire.foo);
    });
</script>
```

What am I doing wrong here?

**Comment by joshhanley** [comment]
@jochemgruter `$wire` is an Alpine directive, so won't work outside of Alpine. Try using `@this` inside your component. But in your layout file like your example you need to use `Livewire` see here https://laravel-livewire.com/docs/2.x/lifecycle-hooks#js-hooks

Hope this helps!

**Comment by oliverbj** [comment]
I had this error too, and simply fixed it by applying `this`:

```javascript
<script>
    document.addEventListener('livewire:load', function () {
        console.log(this.$wire.foo);
    });
</script>
```

**Reply by brandon-9** [reply]
THANK YOU!

**Reply by devsquad-victor-lima** [reply]
Thanks bro, this saved me.

**Comment by TwilightDuck** [comment]
I know this is an old topic but for anyone who stumbles on this.
Try calling `window.Livewire.start()` before `Alpine.start()`

Like so:

```
if (window.Livewire) {
    window.Livewire.start();
}

Alpine.start();
```

**Reply by worldwayii** [reply]
This hasn't worked for me with Livewire3. Following https://livewire.laravel.com/docs/uploads#javascript-upload-api for example, `  $wire.upload('photo', file, (uploadedFilename) => {}` in my component and $wire is saying not defined

**Reply by otidh** [reply]
Replacing `$wire.upload` with `@this.upload` works for me.

**Reply by dym-sh** [reply]
even more general, [`Livewire.upload`](https://laravel-livewire.com/docs/2.x/events#in-js) should work for Livewire 2.x and 3.x

**Reply by lucas-dos-santos-silva** [reply]
in my case I had to trigger the event within alpine's x-data, what solved it was using the $dispatch directive directly,       
     ` $dispatch('dataCpf', [formatted]);`
